### PR TITLE
add postgres config and mongo docker compose

### PR DESCRIPTION
**Pull request by DorianKodelja-ecoact**

### CODE REVIEW

**Review by jyates0eco — APPROVED** [review]

File: docker-compose.mongo.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+version: '3.8'
+
+services:
+  # BEGIN keycloak
```
Comment (tepelbaum) [inline docker-compose.mongo.yml:4]: mongo :)

File: docker-compose.mongo.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      - app-network
+      # Use the public network created to be shared between Traefik and
+      # any other service that needs to be publicly available with HTTPS
+      - traefik-network
+
+networks:
+  app-network:
+    external: true
+  traefik-network:
+    external: true
```
Comment (tepelbaum) [inline docker-compose.mongo.yml:22]: To be removed :)

File: docker-compose.yml [hunk 3]
```diff
@@ -7,10 +7,11 @@ services:
     restart: always
     container_name: db
     healthcheck:
-      test: ["CMD-SHELL", "pg_isready"]
+      test: [ "CMD-SHELL", "pg_isready" ]
       interval: 10s
       timeout: 5s
       retries: 5
+    command: postgres -c max_connections=200 -c idle_session_timeout=150000000
```
Comment (tepelbaum) [inline docker-compose.yml:14]: :heart:

Comment (tepelbaum) [reply docker-compose.yml:14]: Curiosity: We cannot pass them as env var ?

File: docker-compose.yml [hunk 4]
```diff
@@ -41,8 +42,7 @@ services:
       - traefik.http.routers.pgadmin-https.tls.certresolver=le
       # https-redirect middleware creation to redirect HTTP to HTTPS
       - traefik.http.middlewares.https-redirect.redirectscheme.scheme=https
-      - traefik.http.middlewares.https-redirect.redirectscheme.permanent=true
-       # Plug middleware to redirect HTTP to HTTPS
+      - traefik.http.middlewares.https-redirect.redirectscheme.permanent=true # Plug middleware to redirect HTTP to HTTPS
```
Comment (tepelbaum) [inline docker-compose.yml:45]: you can leave as it was before :smile:

Comment (DorianKodelja-ecoact) [reply docker-compose.yml:45]: oupsie

File: docker-compose.mongo.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    image: mongo
+    restart: always
+    container_name: mongo_db
+    environment:
+      MONGO_INITDB_ROOT_USERNAME: $mongo_username
+      MONGO_INITDB_ROOT_PASSWORD: $mongo_password
+
+    networks:
+      - app-network
+
```
Comment (tepelbaum) [inline docker-compose.mongo.yml:15]: # END mondo-db
# BEGIN mongo-admin

File: docker-compose.mongo.yml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      ME_CONFIG_MONGODB_ADMINUSERNAME: $mongo_username
+      ME_CONFIG_MONGODB_ADMINPASSWORD: $mongo_password
+      ME_CONFIG_MONGODB_SERVER: mongo_db
+      ME_CONFIG_MONGODB_PORT: "27017"
+
+    labels:
+      # Enable Traefik for this specific "backend" service
+      - traefik.enable=true
+      # Define the port inside of the Docker service to use
+      - traefik.http.services.mongoexpress.loadbalancer.server.port=80
```
Comment (tepelbaum) [inline docker-compose.mongo.yml:29]: It is not 27017 here?

File: docker-compose.mongo.yml [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    image: mongo-express:latest
+    container_name: mongo_admin
+    environment:
+      ME_CONFIG_MONGODB_ADMINUSERNAME: $mongo_username
+      ME_CONFIG_MONGODB_ADMINPASSWORD: $mongo_password
+      ME_CONFIG_MONGODB_SERVER: mongo_db
+      ME_CONFIG_MONGODB_PORT: "27017"
+
+    labels:
+      # Enable Traefik for this specific "backend" service
```
Comment (tepelbaum) [inline docker-compose.mongo.yml:26]:
```suggestion
      # Enable Traefik for this specific "mongoexpress" service
```

File: docker-compose.yml [hunk 8]
```diff
@@ -38,11 +39,9 @@ services:
       - traefik.http.routers.pgadmin-https.rule=Host(`${pgadmin_url}`)
       - traefik.http.routers.pgadmin-https.tls=true
       # Use the "le" (Let's Encrypt) resolver
-      - traefik.http.routers.pgadmin-https.tls.certresolver=le
-      # https-redirect middleware creation to redirect HTTP to HTTPS
+      - traefik.http.routers.pgadmin-https.tls.certresolver=le # https-redirect middleware creation to redirect HTTP to HTTPS
       - traefik.http.middlewares.https-redirect.redirectscheme.scheme=https
-      - traefik.http.middlewares.https-redirect.redirectscheme.permanent=true
-       # Plug middleware to redirect HTTP to HTTPS
+      - traefik.http.middlewares.https-redirect.redirectscheme.permanent=true # Plug middleware to redirect HTTP to HTTPS
```
Comment (tepelbaum) [inline docker-compose.yml:44]: Tu peux remettre comme avant si tu as le courage :)

File: docker-compose.mongo.yml [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+      - traefik.http.middlewares.https-redirect.redirectscheme.scheme=https
+      - traefik.http.middlewares.https-redirect.redirectscheme.permanent=true
+      # Middleware to redirect HTTP to HTTPS
+      - traefik.http.routers.mongoexpress-http.middlewares=https-redirect
+    networks:
+      - app-network
+      - traefik-network
+    depends_on:
+      - mongo_db
+
```
Comment (tepelbaum) [inline docker-compose.mongo.yml:51]: # END mongo-admin

**Pull request merged**